dotCMS/src/main/java/com/dotmarketing/cms/login/factories/LoginFactory.java

```diff
@@ -38,68 +44,46 @@
+                    return true;
                 }
-                return doLogin(userName, null, true, request, response, true);
-            } catch (Exception e) { // $codepro.audit.disable logExceptions
-        		SecurityLogger.logInfo(LoginFactory.class,"An invalid attempt to login (No user found) from IP: " + request.getRemoteAddr() + " :  " + e );
-
-            	return false;
-            }
-        } catch (Exception e) {
-    		SecurityLogger.logInfo(LoginFactory.class,"Auto login failed (No user found) from IP: " + request.getRemoteAddr() + " :  " + e );
-
-            if(useCASLoginFilter){
-            	String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
-            	Logger.info(LoginFactory.class, "Try to retrieve user from LDAP/CAS with id: " + decryptedId);
-            	User newUser = CASAuthUtils.loadUserFromLDAP(decryptedId);
-            	
-            	if(UtilMethods.isSet(newUser)){
-            		User user = null;
-            		Company comp = com.dotmarketing.cms.factories.PublicCompanyFactory.getDefaultCompany();
-            		try {
-						if (comp.getAuthType().equals(Company.AUTH_TYPE_EA)) {
-							user = APILocator.getUserAPI().loadByUserByEmail(decryptedId, APILocator.getUserAPI().getSystemUser(), false);
-						} else {
-							user = APILocator.getUserAPI().loadUserById(decryptedId, APILocator.getUserAPI().getSystemUser(), false);
-						}
-                        
-	            		String userIdFromCAS = (String)request.getSession(false).getAttribute("edu.yale.its.tp.cas.client.filter.user");
-	            		
-						if(UtilMethods.isSet(userIdFromCAS)){
-							CASAuthUtils.setUserValuesOnSession(user, request, response, true);
-						}
-
-                        return true;
-                         
-                    } catch (Exception ex) {
-                    	return false;
+                
+                /*Custom Code*/
+                User user = null;
+                if(Validator.isEmailAddress(decryptedId))
+                    user = APILocator.getUserAPI().loadByUserByEmail(decryptedId,APILocator.getUserAPI().getSystemUser(),false);
+                 else
+                    user = APILocator.getUserAPI().loadUserById(decryptedId,APILocator.getUserAPI().getSystemUser(),false);
+                /* End of Custom Code */
+                try {
+                    String userName = user.getEmailAddress();
+                    Company comp = com.dotmarketing.cms.factories.PublicCompanyFactory.getDefaultCompany();
+                    if (comp.getAuthType().equals(Company.AUTH_TYPE_ID)) {
+                    	userName = user.getUserId();
                     }
-            	}
-            	else
-            		Logger.info(LoginFactory.class, "Unable to retrieve user from LDAP/CAS with id: " + decryptedId);
-            }
+                    return doLogin(userName, null, true, request, response, true);

```

Comment:
I think we need just sync this call here, right< I mean the code before it just getter from database (I think)

Reply:
yes, we were discussing that. But at the end, we thought we could keep it this way